crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+    /// t is the normalized position druation.
+    ///
+    /// In this case, we use the resulting spot price after a hypothetical long
+    /// for `base_amount` is opened.
+    fn rate_after_long(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: Option<FixedPoint>,
+    ) -> Result<FixedPoint> {
+        let resulting_price = self.calculate_spot_price_after_long(base_amount, bond_amount)?;
+        Ok((fixed!(1e18) - resulting_price)
+            / (resulting_price * self.annualized_position_duration()))
+    }
+
+    /// The derivative of the equation for calculating the rate after a long.
+    ///
+    /// For some $r = (1 - p(x)) / (p(x) * t)$, where $p(x)$
+    /// is the spot price after a long of `delta_base`$= x$ was opened and $t$
+    /// is the annualized position duration, the rate derivative is:
+    ///
+    /// $$
+    /// r'(x) = \frac{(-p'(x) p(x) t - (1 - p(x)) (p'(x) t))}{(p(x) t)^2} //
+    /// r'(x) = \frac{-p'(x)}{t p(x)^2}
+    /// $$
+    ///
+    /// We return $-r'(x)$ because negative numbers cannot be represented by FixedPoint.
+    fn rate_after_long_derivative_negation(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        let price = self.calculate_spot_price_after_long(base_amount, Some(bond_amount))?;
+        let price_derivative = self.price_after_long_derivative(base_amount, bond_amount)?;
+        // The actual equation we want to represent is:
+        // r' = -p' / (t p^2)
+        // We can do a trick to return a positive-only version and
+        // indicate that it should be negative in the fn name.
+        Ok(price_derivative / (self.annualized_position_duration() * price.pow(fixed!(2e18))))
+    }
+
+    /// The derivative of the price after a long.
+    ///
+    /// The price after a long that moves shares by $\Delta z$ and bonds by $\Delta y$
+    /// is equal to $P(\Delta z) = \frac{\mu (z_e + \Delta z)}{y - \Delta y}^T$,
+    /// where $T$ is the time stretch constant and $z_e$ is the initial effective share reserves.
+    /// Equivalently, for some amount of `delta_base`$= x$ provided to open a long,
+    /// we can write:
+    ///
+    /// $$
+    /// p(x) = \frac{\mu (z_e + \frac{x}{c} - g(x) - \zeta)}{y_0 - Y(x)}^{T}
+    /// $$
+    /// where $g(x)$ is the [open_long_governance_fee](long::fees::open_long_governance_fee),
+    /// $Y(x)$ is the [long_amount](long::open::calculate_open_long), and $\zeta$ is the
+    /// zeta adjustment.
+    ///
+    /// To compute the derivative, we first define some auxiliary variables:
+    /// $$
+    /// a(x) = \mu (z_e + \frac{x}{c} - g(x) - \zeta) \\
+    /// b(x) = y_0 - Y(x) \\
+    /// v(x) = \frac{a(x)}{b(x)}
+    /// $$
+    ///
+    /// and thus $p(x) = v(x)^T$. Given these, we can write out intermediate derivatives:
+    ///
+    /// $$
+    /// a'(x) = \frac{\mu}{c} - g'(x) \\
+    /// b'(x) = -Y'(x) \\
+    /// v'(x) = \frac{b a' - a b'}{b^2}
```

Comment:
Nit: I get what you're saying by eliding the arguments, but I think it would be clearer to use the following

```suggestion
    /// v'(x) = \frac{b(x) a'(x) - a(x) b'(x)}{b(x)^2}
```